Blog

peteryates opened the pull request:
### Trello card

### Context

Prepare for the blog to shift from Medium to the site.

### Changes proposed in this pull request

Add a blog controller and some functionality to filter pages by path and filter them based on date. This will give us the ability to queue future posts and have them automatically appear on the site when the day is reached.

Currently we're using IS0 8601 dates. YAML doesn't support dates but 8601 gives us the ability to filter and sort without parsing them. I'll add a content test to ensure dates in blog posts are valid.

### Tasks

* [x] Create a blog controller
* [x] Allow blog posts to be selected from the `Pages::Frontmatter` object.
* [x] Add a content test to ensure blog posts have dates in the right format
* [x] Style the blog index and post templates properly
* [x] Finalise the frontmatter format for blog posts

### Final bits

* [x] Should we display the author names on the blog index page?
* [ ] Where will 'Blog' fit in the nav?
* [ ] Should the blog post titles be links too, or should we rely on the 'Read more' link below?
* [ ] Would it be a good idea to display the counts of posts for each of the popular tags?
* [ ] Is "Get Into Teaching Blog" a good title for the main post listing page?

### Preview

| Index | Post |
| ------ | ------- |
| ![Screenshot 2021-06-30 at 11-24-21 Get Into Teaching ](https://user-images.githubusercontent.com/128088/123951875-2b087700-d99d-11eb-87c4-d018467679dd.png) | ![Screenshot 2021-06-30 at 11-27-28 Get Into Teaching Teachers shaping their students' wellbeing](https://user-images.githubusercontent.com/128088/123945637-46bc4f00-d996-11eb-8fa7-82e8c724bdbc.png) |

### Tags listing

| Expanded | Wrapped |
| ------ | -------- |
| ![Screenshot from 2021-07-02 16-38-30](https://user-images.githubusercontent.com/128088/124298378-17534100-db54-11eb-8ab5-80f274921da9.png) | ![Screenshot from 2021-07-02 16-38-40](https://user-images.githubusercontent.com/128088/124298393-1b7f5e80-db54-11eb-8b0a-3815faee800b.png) |


### Post footer

| Desktop | Mobile |
| -------- | ------- |
| ![Screenshot from 2021-07-05 14-04-58](https://user-images.githubusercontent.com/128088/124475826-13b0fb80-dd9a-11eb-8092-59ef5fe1d261.png) | ![Screenshot from 2021-07-05 14-05-06](https://user-images.githubusercontent.com/128088/124475841-17dd1900-dd9a-11eb-9b10-80515a5274ac.png) |

### Footnotes

Some posts have a footnote. Here's how they look and work using Kramdown's default footnote behaviour.


https://user-images.githubusercontent.com/128088/124486623-b28f2500-dda5-11eb-91c7-d374dffffd04.mp4

gazcarr commented:
**Homepage**

- Popular blog tags should all be clickable and have one single hover state
- Article Headings are too small - should be equiv of 36px
- Published date should be prefixed with "Published" in italics
- Published date and author on wide screens should sit to the right of the heading
- At the bottom of each article there should be a 1px divide with 50px spacing above and below
(if there is no pagination then we can remove the divide on the last article in the list)

**Blog tags listing page**

Can we show other available (or popular/related) tags as per design?

**Article page**

- Author and published date should be on separate lines
- 'Published' and 'by' should be in italics
- No need for the divide below the date/author

peteryates commented:
> Can we show other available (or popular/related) tags as per design?

Which design? I was purely working from this one: https://xd.adobe.com/view/60d5c8ab-ec11-4cff-ae88-149c956fe704-3586/screen/e27db4b8-a643-4884-b316-a8e097d01d0c

peteryates commented:
@gazcarr I addressed some of the issues above. The CSS should be easy to tweak for anything that's remaining, not sure the behaviour of the common tags was clear from the design i referenced above.
